future-apps/payroll/app/src/panels/AddEmployee.js

```diff
@@ -0,0 +1,206 @@
+            type: 'string'
+          },
+          role: {
+            type: 'string'
+          },
+          accountAddress: {
+            type: 'string',
+            format: 'address'
+          }
+        },
+        required: ['accountAddress']
+      }
+    },
+    required: ['salary', 'startDate', 'entity']
+  })
+
+  componentDidUpdate (prevProps, prevState) {
+    if (this.state !== prevState) {
+      const state = { ...this.state }
+
+      this.setState({
+        ...state,
+        isValid: AddEmployee.validate(state)
+      })
+    }
+  }
+
+  focusFirstEmptyField = () => {
+    const { entity, salary } = this.state
+
+    if (!entity) {
+      this.entitySearch.input.focus()
+    } else if (!salary) {
+      this.salaryInput.focus()
+    }
+  }
+
+  handleEntityChange = (accountAddress, entity) => {
+    this.setState({ entity }, () => {
+      this.focusFirstEmptyField()
+    })
+  }
+
+  handleFormSubmit = (event) => {
+    event.preventDefault()
+
+    const { app } = this.props
+
+    if (app) {
+      const accountAddress = this.state.entity.accountAddress
+      const initialDenominationSalary = this.state.salary / SECONDS_IN_A_YEAR
+      const name = this.state.entity.domain
+      const startDate = Math.floor(this.state.startDate.getTime() / 1000)
+
+      app.addEmployeeWithNameAndStartDate(
+        accountAddress,
+        initialDenominationSalary,
+        name,
+        startDate
+      ).subscribe(employee => {
+          if (employee) {
+            console.info(`Created employee ${employee}`)
+
+            // Reset form data
+            this.setState(AddEmployee.initialState)
+
+            // Close side panel
+            this.props.onClose()
+          }
+        }
+      )
+    }
+  }
+
+  handleSalaryChange = (event) => {
+    this.setState({ salary: event.target.value })
+  }
+
+  handleStartDateChange = (date) => {
+    this.setState({ startDate: date })
+  }
+
+  handlePanelToggle = (opened) => {
+    if (opened) { // When side panel is shown
+      this.focusFirstEmptyField()
+    }
+  }
+
+  render () {
+    const { entity, salary, startDate, isValid } = this.state
+
+    return (
+      <SidePanel
+        title='Add new employee'
+        opened={this.props.opened}
+        onClose={this.props.onClose}
+        onTransitionEnd={this.handlePanelToggle}
+      >
+        <Form onSubmit={this.handleFormSubmit}>
+          <Field label='Entity'>
+            <Input.Entity
+              ref={el => this.entitySearch = el}
+              key={entity && entity.domain}
+              value={entity && entity.domain}
+              onChange={this.handleEntityChange}
+            />
+          </Field>
+
+          <Field label='Salary'>
+            <Input.Currency
+              innerRef={el => this.salaryInput = el}
+              value={salary || ''}
+              onChange={this.handleSalaryChange}
+            />
+          </Field>
+
+          <Field label='Start Date'>
+            <Input.Date
```

Comment:
Same as with `Icon.Currency` on with calendar icon. I'm not sure what the best approach here is, I guess either:

1. Make a calendar and $ icon for these fields or,
2. Use a placeholder for now and wait until icons are added to the aragon-ui repo

Reply:
That feature will be implemented in next milestone